ql/src/java/org/apache/hadoop/hive/ql/plan/mapper/PlanMapper.java

```diff
@@ -286,8 +271,13 @@ public Iterator<EquivGroup> iterateGroups() {
   }
 
   public OpTreeSignature getSignatureOf(Operator<?> op) {
-    OpTreeSignature sig = signatureCache.getSignature(op);
-    return sig;
+    return signatureCache.getSignature(op);
+  }
+
+  private void registerSignature(Object o) {
+    if (o instanceof Operator) {
+      getSignatureOf((Operator<?>) o);
+    }
   }
```

Comment:
whats the point of this method? the cache will compute it if needed...what the point of pre-registering - if that's needed its not a *cache* anymore!